### PR TITLE
Checkout: Thank You: Use `Gridicon` instead of noticons

### PR DESCRIPTION
**Pull request by drewblaisdell**
This PR:

- Updates the icons on `CheckoutThankYou` per @breezyskies' spreadsheet.
- Updates `PurchaseDetail` to accept an `icon` prop that is used to mount a `Gridicon`, instead of using noticons in CSS.

**Testing**

We should assert that the icons on the thank you page render correctly after purchasing the following items:

```
business-plan
chargeback
domain-mapping
domain-registration
google-apps
jetpack-business-plan
jetpack-premium-plan
premium-plan
site-redirect
```

- [x] Code review
- [ ] Product review

### CODE REVIEW

**Comment by drewblaisdell** [comment]
We can't test the Jetpack-specific icons here until [this PR](https://github.com/Automattic/wp-calypso/pull/3617) is merged.

**Comment by scruffian** [comment]
Some issues I noticed:
1. The back to my site button looks broken:
screen shot 2016-02-29 at 13 18 43

2. The icons are grey on a white background, not white on a grey background:
screen shot 2016-02-29 at 13 19 11

**Comment by scruffian** [comment]
I updated ec7dcf4 to fix the bugs I noticed above ^

@breezyskies @mikeshelton1503  this could do with a design review.

File: client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx [hunk 1]
```diff
@@ -8,22 +8,24 @@ import React from 'react';
...
 	title
 } ) => {
-	const classes = classNames( 'checkout-thank-you__purchase-detail', additionalClass, {
+	const classes = classNames( 'checkout-thank-you__purchase-detail', {
 		'is-placeholder': isPlaceholder
 	} );
 
 	return (
 		<div className={ classes }>
+			{ icon && <div className="gridicon-wrapper"><Gridicon icon={ icon } size={ 72 } /></div> }
```
Comment (mtias) [inline client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx]: This class should be namespaced to the component: `checkout-thank-you__gridicon-wrapper`.

File: client/my-sites/upgrades/checkout-thank-you/style.scss [hunk 2]
```diff
@@ -74,86 +74,11 @@
 		border-radius: 3px;
 		max-width: 700px;
 		margin: 16px auto;
-		padding-right: 130px;
+		padding-right: 140px;
 		text-align: left;
 	}
 
-	&::before {
```
Comment (mtias) [inline client/my-sites/upgrades/checkout-thank-you/style.scss]: Great to see this :)

**Comment by mtias** [comment]
This is cool. I think `purchase-detail.jsx` could be its own component outside of `checkout-thank-you` by the way. Something like `<PlanFeatureReminder>` or something like that.

**Comment by breezyskies** [comment]
A couple sizing tweaks for the icons:

Desktop:
- Icon size: 48x48
- `padding: 16px`

Mobile (<660px):
- Icon size: 32x32
- `padding: 16px`

**Comment by fabianapsimoes** [comment]
Are we not showing the "Back to my site" label for mobile? This is how it looks now:

screen shot 2016-02-29 at 17 35 58

**Comment by fabianapsimoes** [comment]
Some gridicons look quite misaligned still. For example:

screenshot-calypso localhost 3000 2016-02-29 17-45-10

**Comment by scruffian** [comment]
> Icon size: 32x32

@breezyskies - is it ok to show Gridicons at 32px? They generally ask that we show them at a multiple of 24...

**Comment by fabianapsimoes** [comment]
Seems we're missing an icon here, for Google Apps purchases:
screenshot-calypso localhost 3000 2016-02-29 17-54-43

**Comment by breezyskies** [comment]
> is it ok to show Gridicons at 32px? They generally ask that we show them at a multiple of 24...

Let's go with 36px, per [this documentation](https://wpcalypso.wordpress.com/devdocs/client/components/gridicon/README.md?term=gridicons).

File: client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx [hunk 3]
```diff
@@ -8,22 +8,24 @@ import React from 'react';
...
 	title
 } ) => {
-	const classes = classNames( 'checkout-thank-you__purchase-detail', additionalClass, {
+	const classes = classNames( 'checkout-thank-you__purchase-detail', {
 		'is-placeholder': isPlaceholder
 	} );
 
 	return (
 		<div className={ classes }>
+			{ icon && <div className="checkout-thank-you__gridicon-wrapper"><Gridicon icon={ icon } size={ 72 } /></div> }
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/purchase-detail.jsx]: @scruffian We should namespace this class under the subcomponent namespace, like: `checkout-thank-you__purchase-detail-gridicon-wrapper`.

File: client/my-sites/upgrades/checkout-thank-you/style.scss [hunk 4]
```diff
@@ -183,4 +108,30 @@
 			}
 		}
 	}
+
+	.checkout-thank-you__gridicon-wrapper {
```
Comment (drewblaisdell) [inline client/my-sites/upgrades/checkout-thank-you/style.scss]: @scruffian We don't need to include this selector inside the `.checkout-thank-you__purchase-detail` ruleset.

**Comment by drewblaisdell** [comment]
I made some changes:

- Updated the icon sizes per @breezyskies' note.
- Fixed the icon for Google Apps (the icon slug was `email` instead of `mail`).
- Moved `PurchaseDetail` to `client/components/` per @mtias' recommendation, and updated the CSS accordingly. I kept the name `PurchaseDetail`, instead of the suggested `PlanFeatureReminder`, because it is simpler and because most of our purchase items are not plans.

> Are we not showing the "Back to my site" label for mobile? This is how it looks now:

Not related to this PR specifically, but the default behavior of `HeaderCake` is to hide text from the `backText` prop ("Back to my site", in this case) on mobile viewports.

> Some gridicons look quite misaligned still. For example:

I'm not seeing any more misaligned Gridicons after the switch to 48px/36px icons ([clock screenshot](https://cloudup.com/cdIhjDOuB0g)) but please, let me know if you still notice them.

**Comment by breezyskies** [comment]
Icon sizes look good :+1:

**Comment by scruffian** [comment]
Code :+1:

**Comment by fabianapsimoes** [comment]
Product :+1:

**Pull request merged**